modules/core/src/utils/logtagmanager.hpp

```diff
@@ -37,8 +37,13 @@ class LogTagManager
     // also, extensible functions (accepting user-provided callback) are not allowed
     // to call LogTagManger (to prevent iterator invalidation), which needs enforced
     // with a non-recursive mutex.
+#ifndef CV_DISABLE_MULTI_THREAD
     using MutexType = std::mutex;
     using LockType = std::lock_guard<MutexType>;
+#else // CV_DISABLE_MULTI_THREAD
+    using MutexType = cv::Mutex;
+    using LockType = cv::AutoLock;
```

Comment:
Perhaps we can use this branch only with `cv::` classes.

Reply:
Done